src/primitives/orchard_domain.rs

```diff
@@ -0,0 +1,43 @@
+//! Orchard-specific note encryption domain.
+
+// Review hint: this file is largely derived from src/note_encryption.rs
+
+use crate::{
+    action::Action, note::Rho, primitives::compact_action::CompactAction,
+    primitives::orchard_primitives::OrchardPrimitives,
+};
+
+/// Orchard-specific note encryption logic.
+#[derive(Debug, Clone)]
+pub struct OrchardDomain<P: OrchardPrimitives> {
+    /// A parameter needed to generate the nullifier.
+    pub rho: Rho,
+    phantom: core::marker::PhantomData<P>,
+}
```

Comment:
_From str4d ([r2599584010](https://github.com/zcash/orchard/pull/471#discussion_r2599584010))_

Bug: Missing `memuse::DynamicUsage` impl that was not moved from `src/note_encryption.rs`:
```suggestion
}

impl<P: OrchardPrimitives> memuse::DynamicUsage for OrchardDomain<P> {
    fn dynamic_usage(&self) -> usize {
        self.rho.dynamic_usage()
    }

    fn dynamic_usage_bounds(&self) -> (usize, Option<usize>) {
        self.rho.dynamic_usage_bounds()
    }
}
```

Reply:
FIXED